shared/src/main/scala/mlscript/JSBackend.scala

```diff
@@ -459,4 +498,26 @@ object JSBackend {
     case DecLit(value) => s"DecLit($value)"
     case StrLit(value) => s"StrLit($value)"
   }
+
+  /** Make a function for `with` construct.
```

Comment:
Can you document what the JS looks like?

Reply:
Yes. I will document it in Scaladoc comments.